listenstore/listenstore/listenstore.py

```diff
@@ -89,23 +87,31 @@ def insert_postgresql(self, listens):
                 if not listen.validate():
                     raise ValueError("Invalid listen: %s" % listen)
                 try:
-                    res = connection.execute(
-                    """ INSERT INTO listen(user_id, ts, artist_msid, album_msid, recording_msid, raw_data)
-                        VALUES ( %(user_id)s, to_timestamp(%(ts)s), %(artist_msid)s, %(album_msid)s,
-                        %(recording_msid)s, %(raw_data)s) ON CONFLICT DO NOTHING """, self.format_dict(listen))
+                    params = self.format_dict(listen)
+                    res = connection.execute("""
+                        INSERT INTO listen(user_id, ts, artist_msid, album_msid, recording_msid)
+                        VALUES (%(user_id)s, to_timestamp(%(ts)s), %(artist_msid)s, %(album_msid)s,
+                        %(recording_msid)s) ON CONFLICT DO NOTHING RETURNING id """, params)
+                    params['_id'] = res.fetchone()[0]
+
+                    res = connection.execute("""
+                        INSERT INTO listen_json(id, data) VALUES (%(_id)s, %(data)s) ON CONFLICT DO NOTHING
+                    """, params )
                 except Exception, e:     # Log errors
-                        self.log.error(e)
+                    self.log.error(e)
 
     def execute(self, query, params={}):
         with self.engine.connect() as connection:
             res = connection.execute(query, params)
             return res.fetchall()
 
     def fetch_listens_from_storage(self, user_id, from_id, to_id, limit, order, precision):
-        query = """ SELECT id, user_id, extract(epoch from ts), artist_msid, album_msid, recording_msid, raw_data """ + \
-                """ FROM listen WHERE user_id = %(user_id)s """ + \
-                """ AND extract(epoch from ts) > %(from_id)s AND extract(epoch from ts) < %(to_id)s  """ + \
-                """ ORDER BY extract(epoch from ts) """ + ORDER_TEXT[order] + """ LIMIT %(limit)s"""
+        query = """ SELECT listen.id, user_id, extract(epoch from ts), artist_msid, album_msid, recording_msid, data
+                    FROM listen, listen_json
+                    WHERE listen.id = listen_json.id AND user_id = %(user_id)s AND extract(epoch from ts) > %(from_id)s
+                        AND extract(epoch from ts) < %(to_id)s
+                    ORDER BY extract(epoch from ts) """ + ORDER_TEXT[order] + """ LIMIT %(limit)s
+                """
```

Comment:
This query needs to be formatted a bit better. Split long lines, align statements. See https://github.com/metabrainz/acousticbrainz-server/blob/master/db/dataset_eval.py#L68-L73 for example. @alastair might have more suggestions.